src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -167,5 +167,88 @@ private static Args CreateNativeString(string s)
             s = TrunacteLongString(s);
             return new Args(Native.pw_createStringWithLength(s, Convert.ToUInt64(s.Length)));
         }
+
+        public static string FormatArgs(object o)
+        {
+            var sb = new StringBuilder();
```

Comment:
Maybe you can use: https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Datadog.Trace/Util/StringBuilderCache.cs to remove a couple of allocations.